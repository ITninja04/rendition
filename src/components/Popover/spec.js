/* globals expect, describe, it */
import React from 'react'
import renderer from 'react-test-renderer'
import { Box, Popover, Provider, Txt } from '../../../dist'

describe('Popover component', () => {
  it('should match the stored snapshot', () => {
    const TextWithPopover = () => {
      const [target, setTarget] = React.useState(null)

      return (
        <Box m={3}>
          {target && (
            <Popover target={target} placement='right'>
              <Txt align='center'>Popover content</Txt>
            </Popover>
          )}
          <span ref={setTarget}>Popover target</span>
        </Box>
      )
    }

    const component = renderer.create(
      <Provider>
        <TextWithPopover />
      </Provider>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
