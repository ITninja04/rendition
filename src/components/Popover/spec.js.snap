// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover component should match the stored snapshot 1`] = `
.c0 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  line-height: 1.5;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  box-sizing: border-box;
}

.c3 {
  margin: 16px;
}

.c1 {
  font-family: 'Source Sans Pro',Helvetica,sans-serif;
  font-size: 14px;
  color: #2A506F;
}

<div
  className="c0 c1"
>
  <div
    className="c2 c3"
  >
    <span>
      Popover target
    </span>
  </div>
</div>
`;
